frontend/components/Cell.js

```diff
@@ -117,16 +120,129 @@ export const Cell = ({
     const class_code_folded = code_folded && cm_forced_focus == null
 
     let show_input = errored || class_code_differs || !class_code_folded
+    const prepareFileBase64 = (file) =>
+        new Promise((resolve, reject) => {
+            const { name, type } = file
+            const fr = new FileReader()
+            fr.onerror = () => reject("Failed to read file!")
+            fr.onloadstart = () => {}
+            fr.onprogress = ({ loaded, total }) => {}
+            fr.onload = () => {}
+            fr.onloadend = ({ target: { result } }) => resolve({ fileBase64: String(result).replace(/.*base64,/, ""), name, type })
+            fr.readAsDataURL(file)
+        })
 
+    const uploadAndCreateCodeTemplate = async (file) => {
+        if (!(file instanceof File)) return " #  File can't be read"
+        setSavingFile(true)
+
+        const {
+            message: { success, file_path, file_name },
+        } = await prepareFileBase64(file).then(
+            (preparedObj) => {
+                return requests.write_file(cell_id, preparedObj)
+            },
+            () => alert("Pluto can't save this file 😥")
+        )
+        setSavingFile(false)
+        if (!success) {
+            alert("Pluto can't save this file 😥")
+            return "# File save failed"
+        }
+
+        const [name, extension] = file_name.split(".").map((a) => a.replace(/["\-,#@!\%\s+\;()$&*\[\]\{\}']/g, ""))
+        switch (file?.type) {
+            // https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+            case "text/plain":
+                if (extension === "csv")
+                    return `begin
+    using TableIO, DataFrames #
+    df_${name} = DataFrame(read_table(raw"${file_path}"); copycols=false)
+end`
+                if (extension === "txt")
+                    return `begin
+    file = open(raw"${file_path}")
+    txt_${name} = read(file, String)
+end`
+            case "application/vnd.ms-excel":
+            case "application/vnd.oasis.opendocument.spreadsheet":
+            case "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
+                return `begin
+    using TableIO, DataFrames
+    df_${name} = DataFrame(read_table(raw"${file_path}",
+    #, "Sheet1"  # Uncomment This if you need to read a specific sheet
+    ); copycols=false)
+end
+                `
+            case "image/jpeg":
+            case "image/png":
+            case "image/gif":
+                return `begin
+    using PlutoUI
+    LocalResource(raw"${file_path}")
+ end`
+            case "application/vnd.apache.arrow.file":
+                console.log("Arrow file")
+                break
+            case "application/json":
+                return `begin
+    using TableIO, DataFrames
+    df_${name} = read_table(raw"${file_path}") |> DataFrame
+end`
+            default:
```

Comment:
Could you extract the file type specific code part to a separate module so that it is easier to change / enhance?
Maybe this could even be put into a configuration file.

Reply:
Yes that's a good idea. Maybe it can also be part of the backend code too, and passed through the response (which can be useful if the julia part knows what modules are already imported/installed and what need to be `using`ed/`Pkg.add`ed.

Reply:
If this logic is in the backend, I could extract the identification of file formats and mapping to the corresponding libraries as a separate little package (< 100 LOC, without dependencies of itself). Then both Pluto and TableIO can import this, making sure that file type mapping is always consistent and easy to enhance.

Reply:
I like it

Reply:
First version of the file format mapping package:

https://github.com/lungben/TableFileFormats.jl

Reply:
I moved the code to julia - I think that's better suited in a separate module indeed. I didn't use it in the latest update, but let's discuss!

Reply:
I added the sample code generation here:
https://github.com/lungben/TableFileFormats.jl/tree/example_code_gen

Could you please check if this is working for you?

As discussed, TableFileFormats should be renamed to TableIOInterface to reflect the enhanced use case.